app/src/main/java/de/danoeh/antennapod/ui/screen/download/DownloadLogDetailsDialog.java

```diff
@@ -1,71 +1,219 @@
-        super(context);
+    private Runnable dialogDismissFunction;
+
+    /**
+     * Creates a dialog with Feed title (and FeedItem title if possible).
+     */
+    public DownloadLogDetailsDialog(@NonNull Activity activity, DownloadResult status) {
+        this(activity, status, false, null);
+    }
+
+    /**
+     * Creates a dialog with Feed title (and FeedItem title if possible).
+     * The title(s) can be made clickable, so they jump to the Feed if clicked.
+     * A callback can be set that will be called when the dialog is closed.
+     */
+    public DownloadLogDetailsDialog(@NonNull Activity activity, DownloadResult status, boolean isTitleClickable,
+                                    @Nullable Runnable onDismissCallback) {
+        super(activity);
+
+        Feed feed = null;
+        String podcastTitle = null;
+        String episodeTitle = null;
 
         String url = "unknown";
         if (status.getFeedfileType() == FeedMedia.FEEDFILETYPE_FEEDMEDIA) {
             FeedMedia media = DBReader.getFeedMedia(status.getFeedfileId());
-            if (media != null) {
+            if (media != null && media.getItem() != null) {
+                feed = DBReader.getFeed(media.getItem().getFeedId(), false, 0, 0);
+                if (feed != null) {
+                    podcastTitle = feed.getFeedTitle();
+                }
+                episodeTitle = media.getEpisodeTitle();
                 url = media.getDownloadUrl();
             }
         } else if (status.getFeedfileType() == Feed.FEEDFILETYPE_FEED) {
-            Feed feed = DBReader.getFeed(status.getFeedfileId(), false, 0, 0);
+            feed = DBReader.getFeed(status.getFeedfileId(), false, 0, 0);
             if (feed != null) {
+                podcastTitle = feed.getFeedTitle();
                 url = feed.getDownloadUrl();
             }
         }
 
-        String message = context.getString(R.string.download_successful);
+        String message = activity.getString(R.string.download_successful);
         if (!status.isSuccessful()) {
             message = status.getReasonDetailed();
         }
 
-        String humanReadableReason = context.getString(DownloadErrorLabel.from(status.getReason()));
-        SpannableString errorMessage = new SpannableString(context.getString(R.string.download_log_details_message,
+        String humanReadableReason = activity.getString(DownloadErrorLabel.from(status.getReason()));
+        SpannableString errorMessage = new SpannableString(activity.getString(R.string.download_log_details_message,
                 humanReadableReason, message, url));
         errorMessage.setSpan(new ForegroundColorSpan(0x88888888),
                 humanReadableReason.length(), errorMessage.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
 
+        CharSequence downloadMessage = getDownloadMessage(podcastTitle, episodeTitle);
+
+        SpannableString clickableDownloadMessage;
```

Comment:
Hmm, this span handling is getting more and more complex. How about adding an actual layout xml for the dialog that has buttons, TextViews, etc?